script/client

```diff
@@ -138,11 +147,32 @@ else {
 }
 
 $url->path($path);
-$url->query([%params]) if %params;
+
+if (!$options{'form'}) {
+    $url->query([%params]) if %params;
+}
+else {
+    my %form;
+    for (keys %params) {
+        if (/(\S+)\.(\S+)/) {
+            $form{$1}{$2} = $params{$_};
+        }
+        else {
+            $form{$_} = $params{$_};
+        }
+    }
+    %params = %form;
+}
```

Comment:
I wanted to write "got a" but somehow my android auto-corrected and I was too lazy to correct. I know my comment is not really helpful but to me it looks like many indendation levels now. Better just disregard my comment